test/controllers/ATSPAYEDataControllerTest.scala

```diff
@@ -16,57 +16,113 @@
 
 package controller
 
+import akka.actor.ActorSystem
+import akka.stream.{ActorMaterializer, Materializer}
 import akka.util.Timeout
 import controllers.ATSPAYEDataController
-import controllers.auth.{AuthAction, FakeAuthAction, PayeAuthAction}
+import controllers.auth.{FakeAuthAction, PayeAuthAction}
 import models.paye.PayeAtsMiddleTier
 import org.mockito.Matchers.{eq => eqTo, _}
 import org.mockito.Mockito._
 import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.mockito.MockitoSugar
 import play.api.libs.json.Json
-import play.api.mvc.{Request, Result}
+import play.api.mvc.ControllerComponents
 import play.api.test.FakeRequest
 import play.api.test.Helpers.{contentAsJson, stubControllerComponents}
 import services.NpsService
 import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
 import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
 import utils.TestConstants._
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 import scala.concurrent.duration.Duration
 
 class ATSPAYEDataControllerTest extends UnitSpec with MockitoSugar with WithFakeApplication with ScalaFutures {
 
-  val cc = stubControllerComponents()
-  implicit val timeout = new Timeout(Duration.Zero)
+  val cc: ControllerComponents = stubControllerComponents()
+  implicit val timeout: Timeout = new Timeout(Duration.Zero)
+  implicit val ec: ExecutionContext = cc.executionContext
+  implicit val actorSystem: ActorSystem = ActorSystem()
+  implicit val mat: Materializer = ActorMaterializer()
   val request = FakeRequest()
 
   val npsService: NpsService = mock[NpsService]
   val payeAuthAction: PayeAuthAction = FakeAuthAction
 
   class TestController extends ATSPAYEDataController(npsService, payeAuthAction, cc)
 
-  "getAtsData" should {
+  val cy = 2018
+  val cyPlus1 = 2019
```

Comment:
Will these dates break next year?

Reply:
No, it doesn't matter what these are set to, tests will work